core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/scan/csv/CSVScanSourceOpDesc.scala

```diff
@@ -74,13 +74,7 @@ class CSVScanSourceOpDesc extends ScanSourceOpDesc {
       return null
     }
 
-    val (filepath, fileDesc) = determineFilePathOrDatasetFile()
-    val stream =
-      if (filepath != null) {
-        new FileInputStream(new File(filepath))
-      } else {
-        fileDesc.asInputStream()
-      }
+    val stream = FileResolver.open(new URI(fileUri.get)).asInputStream()
```

Comment:
I think this open should be defined on Document.

Reply:
moved to VirtualDocument